### ISSUE TITLE
Result not generated with Saxon 9.9 HE

### ISSUE THREAD

**Issue by wasmachien75** [xspec/xspec#638]
I am having difficulties running testswhen running xspec.bat with Saxon 9.9 HE.

The generated test result is not a well-formed document as expected, but contains the text value of the x:report element.
```

C:\Scripts\xspec\bin>xspec.bat test.xspec
Saxon script not found, invoking JVM directly instead.

Creating Test Stylesheet...


Running Tests...
Testing with SAXON HE 9.9.0.2
Programs
..Episodes


Formatting Report...
Error on line 1 column 1 of test-result.xml:
  SXXP0003: Error reported by XML parser: Content is not allowed in prolog.
org.xml.sax.SAXParseException; systemId: file:/C:/Scripts/xspec/bin/xspec/test-result.xml; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog.

*** Error formatting the report
```

`test-result.xml`  contains the text-value of the x:report element, instead of the x:report element itself.

When setting SAXON_CP to Saxon 9.8.0.12, everything works fine.

I've set up a repo for reproducing: https://github.com/wasmachien75/xspec-bug

**Comment by AirQuick** [comment]
Quick note:
* Not a recent regression of XSpec. Repro on XSpec v0.3 + Saxon-HE 9.9.1.5.
* Not repro on the current master when `xsl:output` in `test_stylesheet.xsl` is commented out. So seems like https://github.com/xspec/xspec/blob/47f1d47506a801dea2dfe8e836346c172fa35b09/src/compiler/generate-xspec-tests.xsl#L84-L86 is not taking effect.
* Could be a Saxon regression https://saxonica.plan.io/issues/2495 ??

**Comment by wasmachien75** [comment]
Indeed, it seems the named output is ignored.
 I will report this on the Saxon board. Thanks!

**Comment by AirQuick** [comment]
@wasmachien75 
@michaelhkay 

Here is a reduced test case. Just refer to this comment would help:

**test.xsl**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet exclude-result-prefixes="#all" version="2.0"
	xmlns:x="http://www.jenitennison.com/xslt/xspec" xmlns:xs="http://www.w3.org/2001/XMLSchema"
	xmlns:xsl="http://www.w3.org/1999/XSL/Transform">

	<xsl:import href="imported.xsl" />

	<xsl:output indent="yes" method="xml" name="x:report" />

	<xsl:template name="main">
		<xsl:result-document format="x:report">
			<x:report>
				<element>text</element>
			</x:report>
		</xsl:result-document>
	</xsl:template>

</xsl:stylesheet>
```

**imported.xsl**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
	<xsl:output encoding="UTF-8" method="text" />
</xsl:stylesheet>
```

Saxon 9.8:
```console
M:\test>java -jar SaxonHE9-8-0-15J\saxon9he.jar -it:main -t -xsl:test.xsl
...
Saxon-HE 9.8.0.15J from Saxonica
Java version 1.8.0_221
...
<?xml version="1.0" encoding="UTF-8"?>
<x:report xmlns:x="http://www.jenitennison.com/xslt/xspec">
   <element>text</element>
</x:report>
Execution time: 21.5712ms
```

Saxon 9.9:
```console
M:\test>java -jar SaxonHE9-9-1-5J\saxon9he.jar -it:main -t -xsl:test.xsl
...
Saxon-HE 9.9.1.5J from Saxonica
Java version 1.8.0_221
...
Processing  (no source document) initial template = main
textExecution time: 24.8892ms
```

**Comment by AirQuick** [comment]
Closing this, because this turned out to be a Saxon bug and we now have a test (#641) that will be triggered when the next maintenance version of Saxon is released.

**Comment by AirQuick** [comment]
@wasmachien75 
On my end, https://saxonica.plan.io/issues/4315 is still reproduced on Saxon 9.9.1.6.

**Comment by michaelhkay** [comment]
If you have a test case that still fails, please supply it.

Getting serialization parameters right is complicated because there are so many different APIs that use the same rather simple underlying code but impose slightly different rules. We solved the problems we were aware of, but it doesn't surprise me in the slightest if there are other use cases that don't work as expected.

Michael Kay
Saxonica

**Comment by AirQuick** [comment]
@michaelhkay 
I just ran the case in https://saxonica.plan.io/issues/4315 as is.
```console
$ cat test.xsl
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="3.0">
    <xsl:output method="text"/>
    <xsl:output name="xml" method="xml"/>
    <xsl:template name="main">
        <xsl:result-document format="xml">
            <xsl:message>
                <xsl:value-of select="system-property('xsl:product-name')"/>
                <xsl:text> </xsl:text>
                <xsl:value-of select="system-property('xsl:product-version')"/>
            </xsl:message>
            <test/>
        </xsl:result-document>
    </xsl:template>
$ java -jar saxon9he.jar test.xsl -it:main
SAXON HE 9.9.1.6
$ java -jar SaxonHE9-8-0-15J/saxon9he.jar test.xsl -it:main
SAXON HE 9.8.0.15
<?xml version="1.0" encoding="UTF-8"?><test/> 
```

**Comment by AirQuick** [comment]
Finally fixed on Saxon 9.9.1.7.
